### PR TITLE
PHPC-1081: Package build failures on Fedora due to 32-bit issues, and it using MongoDB 3.4

### PR DESCRIPTION
**Pull request by derickr**

### CODE REVIEW

File: tests/bson/bson-timestamp_error-003.phpt [hunk 1]
```diff
@@ -10,15 +10,18 @@ echo throws(function() {
 }, 'MongoDB\Driver\Exception\InvalidArgumentException'), "\n";
 
 echo throws(function() {
-    new MongoDB\BSON\Timestamp(-2147483648, 0);
+    /* I realise that "-2147483647 - 1" could be written as "-2147483648", *however*, PHP considers
+     * the latter a floating point number, as it parses "-" and "2147483648" separately, and
+     * "2147483648" doesn't fit in the 32-bit signed range. */
+    new MongoDB\BSON\Timestamp(-2147483647 - 1, 0);
```
Comment (jmikola) [inline tests/bson/bson-timestamp_error-003.phpt]: I this still going to produce an exception with "Expected increment to be an unsigned 32-bit integer, -2147483648 given" on 32-bit systems?

File: tests/utils/tools.php [hunk 2]
```diff
@@ -137,6 +137,25 @@ function LOAD($uri, $dbname = DATABASE_NAME, $collname = COLLECTION_NAME, $filen
     }
 }
 
+function NEEDS_ATLEAST_MONGODB_VERSION($uri, $version) {
+    $manager = new MongoDB\Driver\Manager($uri);
+    $cmd = new MongoDB\Driver\Command(array("buildInfo" => 1));
```
Comment (jmikola) [inline tests/utils/tools.php]: Short array syntax?

File: tests/bson/bson-timestamp_error-003.phpt (hunk 1 above)
Comment (derickr) [reply tests/bson/bson-timestamp_error-003.phpt]: Yes. That's exactly what this test tests :-)

File: tests/apm/overview.phpt [hunk 3]
```diff
@@ -1,6 +1,7 @@
 --TEST--
 PHPC-349: APM Specification
 --SKIPIF--
+<?php if (PHP_INT_SIZE !== 8) { die('skip Only for 64-bit platform'); } ?>
```
Comment (jmikola) [inline tests/apm/overview.phpt]: Why does this test require a 64-bit platform?

Comment (derickr) [reply tests/apm/overview.phpt]: The cursor ID in one of the returned structures is 64-bit, and can't be represented in 32-bits :-/

File: tests/utils/tools.php (hunk 2 above)
Comment (derickr) [reply tests/utils/tools.php]: I can change that, if you really care :-)

File: tests/apm/overview.phpt (hunk 3 above)
Comment (jmikola) [reply tests/apm/overview.phpt]: Looking at Remi's build log, I see:

```
Fatal error: Uncaught MongoDB\Driver\Exception\InvalidArgumentException: Integer overflow detected on your platform: 917 in /builddir/build/BUILD/php-pecl-mongodb-1.4.0~beta1/NTS/tests/apm/overview.php:17
```

If APM can't work on 32-bit, that may be a more general issue to look into. Outside the scope of this PR, though.

File: tests/utils/tools.php (hunk 2 above)
Comment (jmikola) [reply tests/utils/tools.php]: Please do while merging. It'll just save time later if we come across it again and want to revise.

**Comment by remicollet** [comment]
Still have 2 failed tests (1.4.0beta1 + this PR), only with 3.6 on 32-bit

```
MongoDB\Driver\Manager::executeReadCommand()
[tests/manager/manager-executeReadCommand-001.phpt]

004+ Fatal error: Uncaught MongoDB\Driver\Exception\RuntimeException:
Majority read concern requested, but it is not supported by the storage
engine. in
/builddir/build/BUILD/php-pecl-mongodb-1.4.0~beta1/NTS/tests/manager/manager-executeReadCommand-001.php:19

MongoDB\Driver\Server::executeReadCommand()
[tests/server/server-executeReadCommand-001.phpt]

004+ Fatal error: Uncaught MongoDB\Driver\Exception\RuntimeException:
Majority read concern requested, but it is not supported by the storage
engine. in
/builddir/build/BUILD/php-pecl-mongodb-1.4.0~beta1/NTS/tests/server/server-executeReadCommand-001.php:20
```

As 32-bit only supports mmapv1, can probably be skipped

**Comment by remicollet** [comment]
* scratch build of php-pecl-mongodb-1.4.0~beta1-1.remi.src.rpm for rawhide completed (mongodb 3.6)
	http://koji.fedoraproject.org/koji/taskinfo?taskID=24153227

* scratch build of php-pecl-mongodb-1.4.0~beta1-1.remi.src.rpm for f27 completed (mongodb 3.4)
	http://koji.fedoraproject.org/koji/taskinfo?taskID=24153259

Everything looks good now \o/

Thanks @derickr  for this work.

**Pull request merged**